moto/autoscaling/models.py

```diff
@@ -660,13 +747,22 @@ def update_auto_scaling_group(
         termination_policies,
         new_instances_protected_from_scale_in=None,
     ):
+        # TODO: Add MixedInstancesPolicy once implemented.
+        # Verify only a single launch config-like parameter is provided.
+        if launch_config_name and launch_template:
+            raise ValidationError(
+                "Valid requests must contain either LaunchTemplate, LaunchConfigurationName "
+                "or MixedInstancesPolicy parameter."
+            )
+
         group = self.autoscaling_groups[name]
         group.update(
```

Comment:
Please use named parameters here - with this many params, there's a large chance of getting the wrong parameter order